### PR TITLE
Add sitemap generator

### PR DESCRIPTION
**Pull request by differentreality**
- [x] I have read and agree to the [Code of Conduct](https://github.com/danielschuster-muc/potter-db/blob/master/CODE_OF_CONDUCT.md), and have followed the [Contributing guidelines](https://github.com/danielschuster-muc/potter-db/blob/master/CONTRIBUTING.md).

## Pull Request Type

- [X] New feature (either for website or **api**)

## Summary

Add sitemap https://github.com/danielschuster-muc/potter-db/issues/221 

* Add `sitemap_generator` gem
* Add relevant config file `config/sitemap.rb`
  * with entries for books, characters, movies, potions, and spells index paths
* Update robots file

The generated sitemap in `public/sitemap.xml.gz`

Screenshot from 2022-10-19 11-30-29

### CODE REVIEW

**Review by danielschuster-muc — CHANGES_REQUESTED** [review]
Thank you for your contribution to Potter DB. Can you please implement the request changes?

File: backend/config/sitemap.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  # Add '/articles'
+  #
+  #   add articles_path, :priority => 0.7, :changefreq => 'daily'
+  #
+  # Add all articles:
+  #
+  #   Article.find_each do |article|
+  #     add article_path(article), :lastmod => article.updated_at
+  #   end
+
```
[the remark this answers is not in the document]
Comment (differentreality) [reply backend/config/sitemap.rb:33]: Yep that makes sense

**Review by danielschuster-muc — APPROVED** [review]
Looks good. Thank you for your contribution!

**Comment by differentreality** [comment]
@danielschuster-muc do you think it makes sense to add a cron job (eg see https://github.com/javan/whenever) in order to refresh the sitemap every X time period?

**Comment by danielschuster-muc** [comment]
We actually don’t need this as the Sitemap only contains static / non-dynamic generated pages. And those pages won’t change.

**Pull request merged**